### PR TITLE
Remove the duplicated IOCTL between mtd and block

### PR DESCRIPTION
**Pull request by xiaoxiang781216**
## Summary

This patchset contain three change:
- mtd: Replace MTDIOC_PARTINFO with BIOC_PARTINFO
- Revert "mtd: Add MTDIOC_FLUSH IOCTL like MTDIOC_XIPBASE" 
- mtd: Replace MTDIOC_XIPBASE with BIOC_XIPBASE

All changes try to consolidate the same IOCTL in block layer and mtd layer:

1. Simplify the general driver(e.g. partition, flt) implementation
2. Reduce the confusion with the same IOCTL in the different layer
3. Share the same IOCTL between block and mtd if the concept is same(e.g. XIPBASE, FLUSH and PART).
4. The different concept still expose with the different IOCTL(e.g. geometry, erase...).

Note: I created this PR because the recent trend show that more and more IOCTL which duplicate in block and mtd make the userspace and the general layer(e.g. partition and flt) hard to maintain.

## Impact
MTD layer

### CODE REVIEW

File: drivers/mtd/ftl.c [hunk 1]
```diff
@@ -506,31 +506,9 @@ static int ftl_ioctl(FAR struct inode *inode, int cmd, unsigned long arg)
...
-   * different form).
-   */
-
-  if (cmd == BIOC_XIPBASE)
-    {
-      /* The argument accompanying the BIOC_XIPBASE should be non-NULL.  If
-       * DEBUG is enabled, we will catch it here instead of in the MTD
-       * driver.
-       */
+  /* Only one block driver ioctl command is supported by this driver */
```
Comment (gustavonihei) [inline drivers/mtd/ftl.c:509]: This comment should be removed, it no longer holds true.

File: include/nuttx/mtd/mtd.h [hunk 2]
```diff
@@ -45,11 +45,6 @@
                                            *      receive geometry data (see mtd.h)
                                            * OUT: Geometry structure is populated
                                            *      with data for the MTD */
-#define MTDIOC_XIPBASE    _MTDIOC(0x0002) /* IN:  Pointer to pointer to void in
```
Comment (gustavonihei) [inline include/nuttx/mtd/mtd.h:48]: I believe it is worth adding a note here, similar to this one in Linux:
https://elixir.bootlin.com/linux/latest/source/include/uapi/mtd/mtd-abi.h#L152

**Comment by gustavonihei** [comment]
Also, this is a breaking change, so it needs to be added to the Release Notes for the upcoming release.

File: drivers/mtd/ftl.c (hunk 1 above)
Comment (xiaoxiang781216) [reply drivers/mtd/ftl.c]: Done.

File: include/nuttx/mtd/mtd.h (hunk 2 above)
Comment (xiaoxiang781216) [reply include/nuttx/mtd/mtd.h:48]: Done.

**Comment by xiaoxiang781216** [comment]
> Also, this is a breaking change, so it needs to be added to the Release Notes for the upcoming release.

Done, https://cwiki.apache.org/confluence/display/NUTTX/NuttX+10.2.

**Review by gustavonihei — APPROVED** [review]
LGTM

**Pull request merged**